### DISCUSSION TITLE
Tool Description Display During Tool Change

### DISCUSSION THREAD

**Discussion by MarineCorpse** [terjeio/ioSender#297] [Ideas]
Thanks for the awesome piece of software. I am really enjoying using it and learning the world of CNC. I am using semi-automatic tool changes with multiple different tools in one job. When my G code posts from Fusion 360, the G Code contains the tool number and a text description. Is it possible to implement where this tool description could be displayed in the main Grbl page where messagings such as probing status shows up? I am currently having to go back to the program view to ensure I am loading the correct tool. A minor inconvenience, but seems like it could streamline workflow.

**Comment by terjeio** [comment]
It is already shown in the status line provided that there is a message in the gcode (at least with a grblHAL controller):

![image](https://github.com/terjeio/ioSender/assets/20260062/3a392bd8-d07a-4e55-8377-1c128530291c)

The message can be placed before the tool change command but it the best to have it on the same line.

**Reply by MarineCorpse** [reply]
Interesting. Thanks for the reply. I’ll have to dig into my post processor. My tool message shows up on the line after the M6 line.
